local-cli/server/util/attachHMRServer.js

```diff
@@ -287,4 +377,28 @@ function time() {
   return `${date.getHours()}:${date.getMinutes()}:${date.getSeconds()}:${date.getMilliseconds()}`;
 }
 
+
+function wrapPotentialPromise (valueOrPromise) {
+  if (valueOrPromise == null) {
+    return Promise.resolve(valueOrPromise)
+  }
+  else if (valueOrPromise.then && typeof valueOrPromise === "function") {
```

Comment:
Should this be checking whether `valueOrPromise.then` is a function rather than `valueOrPromise` directly?